app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopup.kt

```diff
@@ -0,0 +1,610 @@
+     */
+    fun show() {
+        windowManager.addView(this, createLayoutParams())
+    }
+
+    @Composable
+    override fun Content() {
+        val anchorLocation = IntArray(2).apply {
+            anchor.getLocationOnScreen(this)
+        }
+
+        val anchorXCoordMiddle = Pixels(anchorLocation.first() + anchor.width / 2)
+        val indicatorArrowHeight = Pixels(properties.indicatorArrowHeight.toPx())
+
+        val popupBounds = computePopupHorizontalBounds(
+            anchorMiddleXCoord = anchorXCoordMiddle,
+            arrowIndicatorWidth = Pixels(CFRPopupShape.getIndicatorBaseWidthForHeight(indicatorArrowHeight.value)),
+        )
+        val indicatorOffset = computeIndicatorArrowStartCoord(
+            anchorMiddleXCoord = anchorXCoordMiddle,
+            popupStartCoord = popupBounds.startCoord,
+            arrowIndicatorWidth = Pixels(
+                CFRPopupShape.getIndicatorBaseWidthForHeight(
+                    properties.indicatorArrowHeight.toPx()
+                )
+            )
+        )
+
+        Popup(
+            popupPositionProvider = object : PopupPositionProvider {
+                override fun calculatePosition(
+                    anchorBounds: IntRect,
+                    windowSize: IntSize,
+                    layoutDirection: LayoutDirection,
+                    popupContentSize: IntSize
+                ): IntOffset {
+                    // Popup will be anchored such that the indicator arrow will point to the middle of the anchor View
+                    // but the popup is allowed some space as start padding in which it can be displayed such that the
+                    // indicator arrow is exactly at the top-start/bottom-start corner but slightly translated to end.
+                    // Values are in pixels.
+                    return IntOffset(
+                        when (layoutDirection) {
+                            LayoutDirection.Ltr -> popupBounds.startCoord.value
+                            else -> popupBounds.endCoord.value
+                        },
+                        when (properties.indicatorDirection) {
+                            IndicatorDirection.UP -> {
+                                when (properties.overlapAnchor) {
+                                    true -> anchorLocation.last() + anchor.height / 2
+                                    else -> anchorLocation.last() + anchor.height
+                                }
+                            }
+                            IndicatorDirection.DOWN -> {
+                                when (properties.overlapAnchor) {
+                                    true -> anchorLocation.last() - popupContentSize.height + anchor.height / 2
+                                    else -> anchorLocation.last() - popupContentSize.height
+                                }
+                            }
+                        }
+                    )
+                }
+            },
```

Comment:
This takes up a considerable size of the instantiation for the `Popup`. Would it make sense to break this into out into a private function so the instantiation is a little cleaner?